File: backend/src/002-entrada_por_teclado/keyboard_input.rb
#! /usr/bin/ruby

=begin (Esto es un comentario de varias líneas)

Con la función gets en ruby, podemos obtener las entradas por teclado,
pero lo hace en la línea inferior a la petición "Escribe tu nombre: ",
así que creamos la función prompt para solucionarlo de forma simple:

https://stackoverflow.com/questions/2889720/one-liner-in-ruby-for-displaying-a-prompt-getting-input-and-assigning-to-a-var
=end

def prompt(*args)
    print(*args)
    gets
end

name = prompt "Escribe tu nombre: " # Esto recoge "tuNombre/n", recoge la pulsación ENTER
name = name.chomp # Con chomp le quitamos el salto de línea

puts "Bienvenid@ al curso, %s." % name # El formateo de Python!

=begin   
También podría formatearse el saludo anterior así:

puts "Bienvenid@ al curso, #{name}."

De hecho, es mas intuitivo en este caso.
=end

File: backend/src/002-stdin/enter/keyboard_input.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

def main():
    name = input("Escribe tu nombre: ")
    print("Bienvenid@ a mi documentación, %s." % name)
    # Otra opción
    #print("Bienvenid@ al curso {}".format(name))

'''
Si hemos ejecutado directamente este archivo
archivo, __name__ será igual a __main__.

Si hemos importado este archivo desde otro,
__name__ != "__main__"  así que no se ejecutará la función main()

Para más información:
https://es.stackoverflow.com/questions/32165/qu%C3%A9-es-if-name-main
'''

if __name__ == '__main__':
    main()
